File: vscodeExt/14.longest-common-prefix.java
/*
 * @lc app=leetcode id=14 lang=java
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
class Solution {

    // ["aa", "a"]
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 0){
            return "";
        }
        int p = 0;
        for(;p<strs[0].length();p++){
            char c = strs[0].charAt(p);
            boolean flag = false;
            for(int i = 1;i<strs.length;i++){
                if(p>=strs[i].length() || strs[i].charAt(p) != c){
                    flag = true;
                    break;
                }
            }
            if(flag){
                break;
            }
                
    
        }
        return strs[0].substring(0, p);
        
    }
}
// @lc code=end

File: round2/14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python
#
# [14] Longest Common Prefix
#

# @lc code=start
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        # Loop through the first string and check the other strings 
        for i in range(len(strs[0])):
            for string in strs[1:]:
                if i > len(string) - 1 or strs[0][i] != string[i]:
                    return strs[0][:i]
        # If there is no other strings.
        return strs[0]

        
# @lc code=end

